### ISSUE TITLE
Specto Library Integration issue

### ISSUE THREAD

**Issue by DaxxD** [mrknow/filmkodi#438]
First off I would like to say I absolutely love this addon. The library integration has been amazingly useful and really sets it apart from the other addons out there.

My library integration was working perfectly until today. A new show was added to my library called "When Tv Was Awesome" I did not add this myself, I had never heard of this show before. My shows I have added like "House of Cards" the first 10 episodes of it went into the "When Tv Was Awesome" folder instead of my "House of Cards" folder.

I tried clearing cache re-adding my shows to library but it has only gotten worse. Everytime I remove "When Tv Was Awesome" from library it adds itself back and now more and more shows are having their episodes stolen by it.

Any help with this problem would be greatly appreciated!

**Comment by Grossdm** [comment]
I'm wondering whether you've gotten a virus. Assuming that you have an Android TV box, get a reputable antivirus from the Android Market (I can't bring myself to say "Google Play").

Do you have any backups of your Kodi install?
If so, you know what to do.
If not, then looking past this problem I recommend the backup add-on available from the Official Kodi Repo.
It's listed under Program add-ons, and the name is Backup. It allows you to schedule backups, save to popular cloud drives, and other nice features.
In principal you could wipe whatever your system is to a factory state, install Kodi and then the  Backup add-on, then restore from wherever your backup is located.
I have used it to restore 2-3 times, and it works. Don't second guess it during a restore, just follow each step that it asks and you'll be right back to when the backup was made.
Best of luck resolving your current problem.

**Comment by Grossdm** [comment]
Another thought about your current problem.
This could be a newer type of malware that is targeted at Kodi. If it is, my guess is that it may not be detected by an antivirus.
You may find that the fastest and easiest way to get everything right again is to do the unthinkable - wipe and reinstall.
No one likes to hear that, so thankfully you're just reading it. ;-)
Again, good luck.

**Comment by DaxxD** [comment]
It is happening on my Raspberry Pi. I am not really sure if RPi is capable of getting that kind of malware/virus.

Thanks for your input

**Comment by DaxxD** [comment]
I did find a way to get my shows in right folder. If you goto video>files> select tv show and change tv information even if it already shows correct show information it will add the episodes to the right show.

There is still the empty tv show "When tv was awesome" that can't be cleaned or removed from library. Very strange. It must have found a way to get stuck in mysql database

**Comment by Grossdm** [comment]
@DaxxD  If you have House of Cards sorted and cleaning the library didn't remove the empty show, I'm wondering if there is an incorrect/spurious directory that has been incorrectly scanned with this "Awesome" show's metadata.
It may be in a different add-on's directory.
Just some thoughts.

**Comment by DaxxD** [comment]
Ya when adding to library through Specto some weird stuff is happening. I'll try my best to explain.

First off I readded the shows that I had deleted trying to fix this. Wayward Pines correctly made a Wayward Pines Folder on my library but had meta info for "When TV Was Awesome" so i rescanned it to have right meta for Wayward.

Second I added "Last Man Standing" which now had somehow converted season 1 of wayward pines to "Last Man Standing" episodes.

All these problems are isolated to adding to library from Specto addon. Haven't had any other issues with library.

Sometimes it changes meta to "When TV Was Awesome" for whole series. Sometimes its just one of the seasons. One season will be correct show other will be a different show.

Now that I know how to fix it, it is fairly painless. It's just another step everytime I add a show to make sure it's metadata is correct for each season of show(you can't just check main show cause one of the seasons could have wrong metadata while the main show has right)

Hope I did a ok job describing it so it can hopefully be fixed in future release

**Comment by Grossdm** [comment]
@DaxxD 
That metadata/library scanning can be a pain.  I'm no expert with Kodi, and I remember some of the early troubles I had.  One was with House of Cards - it would often get scanned as the UK version.

I think that most of my addons now have the date on the end of the Show/Movie title, which seems to help.  I think some metadata sites (IMDB, etc.) identified it as "House of Cards (US)" so that definitely didn't match well.
I suppose I/we could read [the Kodi Wiki](kodi.wiki) to learn about how it identifies library items.
IIRC, when you ask Kodi to refresh a show's info, it often presents a couple likely matches.

Oh, have you tried the [XEM addon](http://kodi.wiki/view/Add-on:XEM)?  You can select it as the scraper when you "Set content" as TV Shows.

Something else to check: after you have the shows/folders scanned, on a new show use the menu to bring up the "Choose content" dialog.  It should have "Selected folder contains a single TV show" checked/ticked/selected.  Perhaps this was/is related to your problem of having different seasons scanned as a different show.

**Comment by DaxxD** [comment]
@Grossdm 
I haven't tried XEM scraper yet. I will look into it.

That's a good thought on folders not being set to individual TV show. I checked and they were all set properly. Seems odd that one season could be set to wrong show when that is selected

**Comment by BankruptEd** [comment]
What's an XEM scraper do?

**Comment by xxcriticxx** [comment]
TheXEM.de is a site that map episode numbering from multiple sources to their TheTVDB.com counterparts. This scraper let you choose between "scene", AniDB.net and TVRage.com, or the original TheTVDB.com order, as the source for the episode numbering; then fetch the correct data from TheTVDB.com. For more information please visit TheXEM.de

**Comment by BankruptEd** [comment]
I see. Thanks for the info.

**Comment by Grossdm** [comment]
@BankruptEd 
I'm sorry about your financial situation.

**Comment by BankruptEd** [comment]
LMAO. Thanks. Hey, while I have your attention, are you very knowledgeable
about Kodi and Specto in particular? Like do you know how to read an error
log and make sense of it? Or do you know someone who is?

**Comment by Grossdm** [comment]
@BankruptEd 
Well, PoorEd, I'm learning more about Kodi each day with CWH® (Countless Wasted Hours) but I can try to help.  Honestly the add-on that I use most often is SALTS, but that requires more user effort to make it work _really_ well.  I don't know Python, but I can usually figure out something that has caught my attention - again, CWH®.
Do you know how to properly post an error log?
Hint: don't copy/paste the text into a post.  Also, search through any log you post for info that you don't want the NSA and others to read, e.g. passwords.  (Well, the NSA probably doesn't need our passwords for much of its snooping.)

**Comment by BankruptEd** [comment]
Actually my issue resolved itself. Another update came through for Specto
and URL resolver and now all is good.

Yes I figured out how to upload a log just yesterday out of necessity, and
yes I only posted the link. Never occurred to me about sensitive stuff
though. It all looked like Chinese to me lol. No offense intended to the
Chinese 😀. At any rate I don't think I really had any sensitive data in
there.

And I agree, the NSA doesn't need any help from us. They pretty much have
the master key I think lol.

But thanks alot for responding. You and a few others on here were very
helpful and friendly. It's been my experience that noobs like me often
don't get treated very well, but my experience here was quite pleasant.
Take care friend 🙋
